storage/storage.go

```diff
@@ -333,3 +345,45 @@ type Keys struct {
 	// For caching purposes, implementations MUST NOT update keys before this time.
 	NextRotation time.Time
 }
+
+// ConnectorConfig is a configuration that can open a connector.
+type ConnectorConfig interface {
+	Open(logrus.FieldLogger) (connector.Connector, error)
+}
+
+var connectorsConfig = map[string]func() ConnectorConfig{
+	"mockCallback": func() ConnectorConfig { return new(mock.CallbackConfig) },
+	"mockPassword": func() ConnectorConfig { return new(mock.PasswordConfig) },
+	"ldap":         func() ConnectorConfig { return new(ldap.Config) },
+	"github":       func() ConnectorConfig { return new(github.Config) },
+	"gitlab":       func() ConnectorConfig { return new(gitlab.Config) },
+	"oidc":         func() ConnectorConfig { return new(oidcConnector.Config) },
+	"saml":         func() ConnectorConfig { return new(saml.Config) },
+	// Keep around for backwards compatibility.
+	"samlExperimental": func() ConnectorConfig { return new(saml.Config) },
+}
+
+// OpenConnector will parse the connector config and open the connector.
+func OpenConnector(logger logrus.FieldLogger, conn Connector) (connector.Connector, error) {
```

Comment:
Ok will move it there. This method is also used by cmd/dex/serve.go and requires the connectorsConfig variable defined. It does seem redundant to define the connectorsConfig variable in both cmd/dex/config and here, but not really sure how to avoid that since we can't import the main pkg.